src/Symfony/Component/Yaml/Parser.php

```diff
@@ -26,6 +26,7 @@ class Parser
     const TAG_PATTERN = '(?P<tag>![\w!.\/:-]+)';
     const BLOCK_SCALAR_HEADER_PATTERN = '(?P<separator>\||>)(?P<modifiers>\+|\-|\d+|\+\d+|\-\d+|\d+\+|\d+\-)?(?P<comments> +#.*)?';
 
+    private $filename;
```

Comment:
This property must be reset when doing a public call to `parse()`, otherwise subsequent usages of the parser would still report the previous filename.
The other alternative is to reset it to `null` in a `finally` clause in `parseFile`

Reply:
fixed